From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Mon, 17 May 2021 14:47:35 +0200
Subject: fix: implement Print Preview methods in PrintViewManagerBasic

Print Preview code was removed from PrintManager in https://chromium-review.googlesource.com/c/chromium/src/+/2888590
and the removed methods were implemented in HeadlessPrintManager but
not PrintViewManagerBasic, where they are also required. This caused
compilation errors in Electron when we pulled in the relevant CL,
as we make use of PrintViewManagerBasic.

This fixes the issue by implementing the missing methods in
PrintViewManagerBasic.

Upstreamed at https://chromium-review.googlesource.com/c/chromium/src/+/2897572

diff --git a/chrome/browser/printing/print_view_manager_basic.cc b/chrome/browser/printing/print_view_manager_basic.cc
index aa73f19eca8f791efbb85fef1ffc545f89acdd5f..0b014154867507da59fcaf2ab290908ebbdcd80c 100644
--- a/chrome/browser/printing/print_view_manager_basic.cc
+++ b/chrome/browser/printing/print_view_manager_basic.cc
@@ -30,6 +30,30 @@ PrintViewManagerBasic::~PrintViewManagerBasic() {
 #endif
 }
 
+#if BUILDFLAG(ENABLE_PRINT_PREVIEW)
+void PrintViewManagerBasic::UpdatePrintSettings(int32_t cookie,
+                                       base::Value job_settings,
+                                       UpdatePrintSettingsCallback callback) {
+  auto params = mojom::PrintPagesParams::New();
+  params->params = mojom::PrintParams::New();
+  std::move(callback).Run(std::move(params), false);
+}
+
+void PrintViewManagerBasic::SetupScriptedPrintPreview(
+    SetupScriptedPrintPreviewCallback callback) {
+  std::move(callback).Run();
+}
+
+void PrintViewManagerBasic::ShowScriptedPrintPreview(bool source_is_modifiable) {}
+
+void PrintViewManagerBasic::RequestPrintPreview(
+    mojom::RequestPrintPreviewParamsPtr params) {}
+
+void PrintViewManagerBasic::CheckForCancel(int32_t preview_ui_id,
+                                  int32_t request_id,
+                                  CheckForCancelCallback callback) {}
+#endif
+
 #if defined(OS_ANDROID)
 void PrintViewManagerBasic::PdfWritingDone(int page_count) {
   pdf_writing_done_callback_.Run(page_count);
diff --git a/chrome/browser/printing/print_view_manager_basic.h b/chrome/browser/printing/print_view_manager_basic.h
index 8bffa6f964e35dd1e85aa3c372895b374e4464f2..842d0ee73d9c207a0f3658df9cc96822b93f7cb9 100644
--- a/chrome/browser/printing/print_view_manager_basic.h
+++ b/chrome/browser/printing/print_view_manager_basic.h
@@ -24,6 +24,20 @@ class PrintViewManagerBasic
   void PdfWritingDone(int page_count) override;
 #endif
 
+#if BUILDFLAG(ENABLE_PRINT_PREVIEW)
+  // printing::mojom::PrintManagerHost:
+  void UpdatePrintSettings(int32_t cookie,
+                           base::Value job_settings,
+                           UpdatePrintSettingsCallback callback) override;
+  void SetupScriptedPrintPreview(
+      SetupScriptedPrintPreviewCallback callback) override;
+  void ShowScriptedPrintPreview(bool source_is_modifiable) override;
+  void RequestPrintPreview(mojom::RequestPrintPreviewParamsPtr params) override;
+  void CheckForCancel(int32_t preview_ui_id,
+                      int32_t request_id,
+                      CheckForCancelCallback callback) override;
+#endif
+
  private:
   explicit PrintViewManagerBasic(content::WebContents* web_contents);
   friend class content::WebContentsUserData<PrintViewManagerBasic>;
